imap_processing/mag/l0/mag_l0_data.py

```diff
@@ -116,3 +119,46 @@ def __post_init__(self) -> None:
 
         self.PRI_VECSEC = 2**self.PRI_VECSEC
         self.SEC_VECSEC = 2**self.SEC_VECSEC
+
+    def __eq__(self, other: object) -> bool:
```

Comment:
Can't you set `eq=True` in the dataclass decorator to get this automatically for you?